File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

class Solution:
    def getIntersectionNode(self, headA: ListNode, headB: ListNode) -> Optional[ListNode]:
        
    
        def getLength(head):
            length = 0
            while head:
                length += 1
                head = head.next
            return length
        
        len1 = getLength(headA)
        len2 = getLength(headB)
        temp1 = headA
        temp2 = headB
        while len1 > len2:
            temp1 = temp1.next
            len1 -= 1
        while len2 > len1:
            temp2 = temp2.next
            len2 -= 1 
        while temp1 and temp2:
            if temp1 == temp2:
                return temp1
            else:
                temp1 = temp1.next
                temp2 = temp2.next
        return None
            
        

File: 0160-intersection-of-two-linked-lists/0160-intersection-of-two-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode(int x) {
 *         val = x;
 *         next = null;
 *     }
 * }
 */
public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        ListNode lista = headA;
        ListNode listb = headB;
        
        while(lista != listb){
            if(lista != null){
                lista = lista.next;
            } else {
                lista = headB;
            }
            if(listb != null){
            listb = listb.next;
            } else {
                listb = headA;
            }
        }
        return lista;
    }
}